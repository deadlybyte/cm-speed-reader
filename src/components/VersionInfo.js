// @flow
import React from 'react';
import './VersionInfo.css';

const VersionInfo = () => {
  return (
    <span>
      Version - {process.env.REACT_APP_VERSION}
    </span>
  );
};

export default VersionInfo;
