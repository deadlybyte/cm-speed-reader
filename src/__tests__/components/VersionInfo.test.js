import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import VersionInfo from '../../components/VersionInfo';
import packageJson from '../../../package.json';

describe('VersionInfo', () => {
  describe('render', () => {
    it('should render without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<VersionInfo />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('should render as expected', () => {
      process.env.REACT_APP_VERSION = packageJson.version;
      const componentUnderTest = renderer.create(<VersionInfo />)
        .toJSON();
      expect(componentUnderTest).toMatchSnapshot();
    });
  });
});
