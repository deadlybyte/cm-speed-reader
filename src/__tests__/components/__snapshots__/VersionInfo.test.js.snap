// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VersionInfo render should render as expected 1`] = `
<span>
  Version - 
  0.1.0
</span>
`;
